programs/timefn.c

```diff
@@ -95,10 +93,18 @@ UTIL_time_t UTIL_getTime(void)
     /* time must be initialized, othersize it may fail msan test.
      * No good reason, likely a limitation of timespec_get() for some target */
     UTIL_time_t time = UTIL_TIME_INITIALIZER;
+    /* Android < 10 lacks timespec_get but does define TIME_UTC. */
+#if defined(TIME_UTC) && !defined(__ANDROID__)
     if (timespec_get(&time, TIME_UTC) != TIME_UTC) {
         perror("timefn::timespec_get");
         abort();
     }
+#else
```

Comment:
Note : What happens in the following theoretical scenario ? 
- C11, aka `__STDC_VERSION__ >= 201112L`
- but `TIME_UTC` is *not* defined, or `__ANDROID__` is defined
- yet `CLOCK_REALTIME` is not defined either.

I would expect a compilation error on `clock_gettime(CLOCK_REALTIME,` line.

Reply:
A bit unrealistic:
- Missing `TIME_UTC` implies incomplete C11 support
- `__ANDROID__` implies Unix (POSIX) thus `CLOCK_REALTIME`

Fuchsia-based Android can be such but I don't know much.

Reply:
Unexpected, sure, but unfortunately, experience learned us that it unfortunately happens. `C11` without `timespec_get()` is a thing, even if it should not.

So we have 2 strategies : 
- let the problem there, and react later when it happens
- preemptively avoid the situation

It looks to me that option 2 is not too hard to obtain :  
just keep the `&& defined(TIME_UTC) && !defined(__ANDROID__)` part of the test above, make it part of the first `C11` test, using parenthesis.

This way, afterwards, if `(defined(TIME_UTC) && !defined(__ANDROID__))` is `false`, then it necessarily implies that the `#else` is fine.

Reply:
It maybe easier to reason when C11 conditional is inverted i.e., prefer `CLOCK_MONOTONIC`. C standards (unlike C++ standards) evolve very slowly and the result is often meh (e.g., C11 threads that barely anyone uses). The extra portability provided by C11 is offset by extra shenanigans of C11 library implementations.